### ISSUE TITLE
Alias for UPDATE table not found as ancestor element

### ISSUE THREAD

**Issue by mdeweerd** [sqlfluff/sqlfluff#6147]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Unexpected notice because 'p' is found in ancestor "UPDATE":
```plaintext
L: 169 | P: 110 | RF01 | Reference 'p.rowid' refers to table/view not found in
                       | the FROM clause or found in ancestor statement.
```

for

```sql
UPDATE llx_product AS p SET p.stock= (SELECT SUM(ps.reel) FROM llx_product_stock AS ps WHERE ps.fk_product = p.rowid);
```

running

```bash
sqlfluff --dialect mysql --exclude-rules "LT09,LT05,LT01,CP01,CP02,CP04,ST06,AM04,AL05"
```

### Expected Behaviour

sqlfluff should not issue a notice for this / accept 'p' as an ancestor 

### Observed Behaviour

mysql emits a notice while the query is valid.

### How to reproduce

Unexpected notice because 'p' is found in ancestor "UPDATE":
```plaintext
L: 169 | P: 110 | RF01 | Reference 'p.rowid' refers to table/view not found in
                       | the FROM clause or found in ancestor statement.
```

### Dialect

mysql

### Version

sqlfluff, version 3.1.0 
Python 3.9.16

### Configuration

Empty, see CLI

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by riku-d** [comment]
@mdeweerd Please, assign this issue to me under hactoberfest
